debugger/src/debug_adapter.rs

```diff
@@ -636,34 +627,26 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
             },
         };
 
-        if let Some(debug_info) = core_data.debug_info.as_ref() {
-            // Evaluate the static scoped variables.
-            let static_variables =
-                match debug_info.get_stack_statics(&mut core_data.target_core, u64::from(pc)) {
-                    Ok(static_variables) => static_variables,
-                    Err(err) => {
-                        let mut error_variable = probe_rs::debug::Variable::new();
-                        error_variable.name = "ERROR".to_string();
-                        error_variable
-                            .set_value(format!("Failed to retrieve static variables: {:?}", err));
-                        vec![error_variable]
-                    }
-                };
+        let regs = core_data.target_core.registers();
 
-            // Store the static variables for later calls to `variables()` to retrieve.
-            let (static_scope_reference, named_static_variables_cnt, indexed_static_variables_cnt) =
-                self.create_variable_map(&static_variables);
+        let pc = match core_data.target_core.read_core_reg(regs.program_counter()) {
+            Ok(pc) => pc,
+            Err(error) => {
+                return self.send_response::<()>(request, Err(DebuggerError::ProbeRs(error)))
+            }
+        };
 
+        if let Some(debug_info) = core_data.debug_info.as_mut() {
             let current_stackframes =
                 debug_info.try_unwind(&mut core_data.target_core, u64::from(pc));
 
             match self.adapter_type() {
                 DebugAdapterType::CommandLine => {
                     let mut body = "".to_string();
-                    // TODO: Update the code to include static variables.
-                    for frame in current_stackframes {
-                        body.push_str(format!("{}\n", frame).as_str());
+                    for _stack_frame in current_stackframes {
+                        // Iterate all the stack frames, so that `debug_info.variable_cache` gets populated.
                     }
+                    body.push_str(format!("{}\n", debug_info.variable_cache).as_str());
```

Comment:
The variable cache is necessary to support 'lazy' loading of child variables - the primary purpose of the PR. When VSCode / MS DAP client requests the child variable, it sends a single `i64` reference number of the parent it is trying to expand. The variable cache is keyed on this number, and allows storing the essential DWARF references that is necessary to find and resolve the children. I tried my best to document (at the code definition) the `VariableCache` structure and how it is structured. I am totally open to design discussion/advice here, but the existing (pre this PR) stack frames simply didn't have enough information to support the lazy loading of variables from the DWARF.

Reply:
My issue is not the variable cache, it's more the current interface. If we don't need an iterator, then we should use a different interface, for example just a function to create / fill the variable cache.

Reply:
You make a good point. The iterator was there since before me, so I didn't want to dismantle it. That said, it might be useful if at some point we only want to resolve the variables for the first (top) frame in the stack, or maybe one frame at a time, on demand. Do you want me to add a TODO in the code to consider removing the iterator?

Reply:
A todo would be great.